src/coreclr/jit/valuenum.cpp

```diff
@@ -8612,10 +8627,18 @@ bool Compiler::fgValueNumberConstLoad(GenTreeIndir* tree)
     FieldSeq* fieldSeq = nullptr;
     if (varTypeIsIntegral(tree) && GetStaticFieldSeqAndAddress(vnStore, tree->gtGetOp1(), &address, &fieldSeq))
     {
-        assert(fieldSeq->GetKind() == FieldSeq::FieldKind::SimpleStaticKnownAddress);
         CORINFO_FIELD_HANDLE fieldHandle = fieldSeq->GetFieldHandle();
+        ssize_t              byteOffset  = 0;
+        if (fieldSeq->GetKind() == FieldSeq::FieldKind::SimpleStaticKnownAddress)
+        {
+            byteOffset = address - fieldSeq->GetOffset();
+        }
+        else
+        {
+            assert(address == -1);
+            assert(fieldSeq->GetKind() == FieldSeq::FieldKind::SimpleStatic);
+        }
```

Comment:
If I am reading this correctly, `byteOffset` will always be zero for the `SimpleStatic` case, which isn't right.